hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReaderSchemaHandler.java

```diff
@@ -207,17 +213,22 @@ private static String[] getMandatoryFieldsForMerging(HoodieTableConfig cfg, Type
         requiredFields.addAll(Arrays.asList(fields.get()));
       }
     }
-
+    // Add precombine field for event time ordering merge mode.
     if (mergingConfigs.getLeft() == RecordMergeMode.EVENT_TIME_ORDERING) {
       String preCombine = cfg.getPreCombineField();
       if (!StringUtils.isNullOrEmpty(preCombine)) {
         requiredFields.add(preCombine);
       }
     }
-
+    // Add `HOODIE_IS_DELETED_FIELD` field if exists.
     if (dataSchema.getField(HoodieRecord.HOODIE_IS_DELETED_FIELD) != null) {
       requiredFields.add(HoodieRecord.HOODIE_IS_DELETED_FIELD);
     }
+    // Add custom delete key field if exists.
+    if (FileGroupRecordBuffer.hasCustomDeleteConfigs(props, dataSchema)) {
+      requiredFields.add(props.getProperty(DELETE_KEY));
```

Comment:
Could we fetch delete marker configs in schema handler once and pass it to record buffer?

Reply:
Fixed.